### PR TITLE
[RLlib] Fix `TorchMultiCategorical.to_deterministic` to return a multi-dimensional tensor instead of a list.  

### PR DESCRIPTION
**Pull request by simonsays1980**
## Why are these changes needed?

Using a `MultiDiscrete` action space results in RLlib in a `MultiCategorical` distribution. When turning this distribution to a deterministic distribution for inference, a list of `TorchDeterministic` distributions is created. This results than in a sampled list of `tensor`s instead of a single `tensor` with proper dimensions. This PR proposes a fix that
- Turns a `TorchMultiCategorical` distribution into a single `TorchDeterministic` that returns a single `tensor` of proper distribution.
- It works also for `MultiDiscrete` distributions with different categories per dimension, by using padding.

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: rllib/models/torch/torch_distributions.py [hunk 1]
```diff
@@ -498,7 +498,16 @@ def from_logits(
         return TorchMultiCategorical(categoricals=categoricals)
 
     def to_deterministic(self) -> "TorchMultiDistribution":
```
Comment (sven1977) [inline rllib/models/torch/torch_distributions.py:500]:
```suggestion
    def to_deterministic(self) -> "TorchDeterministic":
```

**Review by sven1977 — APPROVED** [review]
LGTM now. Thanks for this fix @simonsays1980 !

**Comment by JiangpengLI86** [comment]
It seems that this implementation cannot handle the LSTM-enabled case. I'm trying to solve #52177
